src/Core/Managed/Shared/Extensibility/Implementation/TaskTimerInternal.cs

```diff
@@ -111,7 +111,7 @@ public void Start(Func<Task> elapsed)
                     TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
                     TaskScheduler.Default);
 
-            CancelAndDispose(Interlocked.Exchange(ref this.tokenSource, newTokenSource));
+            DisposeToken(Interlocked.Exchange(ref this.tokenSource, newTokenSource));
```

Comment:
It looks like we only dispose token in this code but never use it to cancel or manage task lifetime. Should we simplify the approach and use only one token across or no token at all if we do not rely on it?